### PR TITLE
Pirate and idea

### PR DESCRIPTION
**Pull request by Awarua-**

### CODE REVIEW

File: modules/pirate.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            }
+        }
+        else {
+            callback({error:'Whomever the system admin is around here, I demand that they should be fired.'});
+        }
+    });
+};
+
+exports.search = function (query, callback) {
+    //TODO waiting for update to anim to use same api to embed images in fb chat.
```
Comment (Awarua-) [inline modules/pirate.js]: remove comment

File: modules/pirate.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+var require_install = require('require-install'),
+    request = require_install('request'),
+    querystring = require("querystring");
```
Comment (mrkno) [inline modules/pirate.js]: require_install

File: modules/common/reddit.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ var require_install = require('require-install'),
 
 exports.reddit = function(thing, numberOfQueries, callback) {
     reddit.r(thing).limit(numberOfQueries, function(err, req, res){
-        if (err || req !== null || req !== undefined) {
+        if (!err || req !== null || req !== undefined) {
```
Comment (fallaciousreasoning) [inline modules/common/reddit.js]: Should those be ands? If there is no error or the request is null or the request is undefined? That or if there is an error?

File: modules/pirate.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+exports.match = function(text) {
+    return text.startsWith('/pirate');
+};
+
+exports.help = function() {
+    return '/pirate <message> : Translates your message in pirate speak.';
```
Comment (Awarua-) [inline modules/pirate.js]: into pirate speak

File: modules/pirate.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        else {
+            callback({error:'Whomever the system admin is around here, I demand that they should be fired.'});
+        }
+    });
+};
+
+exports.search = function (query, callback) {
+    //TODO waiting for update to anim to use same api to embed images in fb chat.
+    if (query === undefined || query === '') {
+        exports.pirate("I can't do anything with this", callback);
```
Comment (fallaciousreasoning) [inline modules/pirate.js]: You can use !query instead of query === undefined

**Pull request merged**